coralproject/talk
# Toxic Comments Plugin

## What does this PR do?
- Implement toxic comment plugin
- Attach perspective scores to the metadata when posting comments
- Let users reconsider their comment if it contains toxic language.
- Flag comments with toxic language
- Removed `wrapResponse` helper for mutation resolvers and achieve a similar result using the new `errorHandler` schema decorator.
- A comment that is toxic lands in `premod` (will change in a later PR)
- New environment variables (might change in a later PR)
  - `TALK_PERSPECTIVE_API_ENDPOINT` (default https://commentanalyzer.googleapis.com/v1alpha1)
  - `TALK_PERSPECTIVE_API_KEY`
  -  `TALK_TOXICITY_THRESHOLD` (default 0.8)
  - `TALK_PERSPECTIVE_TIMEOUT` (default 300)

## How do I test this PR?

- export TALK_PERSPECTIVE_API_KEY=<api key> # see 1password for test api key
- start Talk
- enter a toxic comment (e.g. one with curse words)
- verify that user is warned 
- finally post the comment, it should not appear in stream (if still toxic)
- Verify toxicity score is attached to comment metadata
